iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -1626,6 +1630,12 @@ + (void)load {
     [OneSignalHelper registerAsUNNotificationCenterDelegate];
 }
 
++(BOOL) shouldDisableBasedOnProcessArguments {
+    if ([NSProcessInfo.processInfo.arguments containsObject: @"DISABLE_ONESIGNAL_SWIZZLING"]) {
```

Comment:
We should check the value of `DISABLE_ONESIGNAL_SWIZZLING`, so if someone sets this to `false` it doesn't still disable it which would be confusing. Most people are going set and forget this flag but someone may come back and set this to false to debug something and it won't behave as expected.

Reply:
hi @jkasten2,  `NSProcessInfo.processInfo.arguments` returns an array and not a dictionary (`NSProcessInfo.processInfo.environment`). The name of the value is self-explanatory, and its confusion to introduce double-negatives with `false`.

I have considered your concern about developers forgetting to remove to the argument. I have added a warning to alert developers.

Reply:
Ah my mistake, you are right `arguments` is an array so presents is the correct way to do this.

Also warning looks good to me!